jxl/src/util/fast_math.rs

```diff
@@ -61,8 +109,51 @@ pub fn fast_powf(base: f32, exp: f32) -> f32 {
 mod test {
     use test_log::test;
 
+    use crate::util::test::assert_almost_eq;
+
     use super::*;
 
+    #[test]
+    fn test_fast_erff() {
+        // Golden data generated by libjxl FastErff.
+        let golden = [
+            -0.796682, -0.741735, -0.677460, -0.603721, -0.520669, -0.428785, -0.328955, -0.222604,
+            -0.111897, 0.000000, 0.111897, 0.222604, 0.328955, 0.428785, 0.520669, 0.603721,
+            0.677460, 0.741735, 0.796682,
+        ];
+        for (index, f_x_10) in (-9..=9).enumerate() {
+            let f = f_x_10 as f32 * 0.1;
+            assert_almost_eq!(fast_erff(f), golden[index], 1e-6);
+        }
+    }
+
+    #[test]
+    fn test_fast_cos() {
```

Comment:
Can we do this comparing against f32::cos() and checking accuracy?

Reply:
I added comparison with f32::cos, and also added a different error function estimation to compare with.